src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/config/WebClientUserEnhancementConfiguration.kt

```diff
@@ -20,54 +18,13 @@ import org.springframework.security.oauth2.client.endpoint.DefaultClientCredenti
 import org.springframework.security.oauth2.client.endpoint.OAuth2ClientCredentialsGrantRequest
 import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
 import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
-import org.springframework.stereotype.Component
+import org.springframework.security.oauth2.server.resource.web.reactive.function.client.ServletBearerExchangeFilterFunction
 import org.springframework.web.context.annotation.RequestScope
-import org.springframework.web.reactive.function.client.ClientRequest
-import org.springframework.web.reactive.function.client.ExchangeFunction
 import org.springframework.web.reactive.function.client.WebClient
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.client.ArnApiClient
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.client.CvlApiClient
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.client.OffenderSearchApiClient
-import java.io.IOException
 import java.net.URI
-import javax.servlet.Filter
-import javax.servlet.FilterChain
-import javax.servlet.FilterConfig
-import javax.servlet.ServletException
-import javax.servlet.ServletRequest
-import javax.servlet.ServletResponse
-import javax.servlet.http.HttpServletRequest
-
-@Component
-@Order(4)
-internal class UserContextFilter : Filter {
-  @Throws(IOException::class, ServletException::class)
-  override fun doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain) {
-    val httpServletRequest = servletRequest as HttpServletRequest
-    val authToken = httpServletRequest.getHeader(HttpHeaders.AUTHORIZATION)
-    authToken?.let {
-      UserContext.setAuthToken(authToken)
```

Comment:
AFAIK, this filter adds user info to the security context. It's used to restrict access e.t.c. to cetain cases.
It's worth Mayank testing limited/restricted cases still work as before once this is merged in.